src/applications/hca/tests/hca.contract.spec.js

```diff
@@ -0,0 +1,101 @@
+import { Pact, Matchers } from '@pact-foundation/pact';
+import { expect } from 'chai';
+
+import environment from 'platform/utilities/environment';
+
+describe('HCA API', () => {
+  const { like, iso8601DateTimeWithMillis } = Matchers;
+
+  const provider = new Pact({
+    port: 3000,
+    consumer: 'VA.gov',
```

Comment:
Maybe we should probably define each app to be a consumer so that we generate individual Pact files for each app.

Reply:
On the other hand, the API should respond to requests to particular endpoints in the same way across apps. Some apps also invoke common endpoints, which do not have particular owners.

Reply:
Is there a difference in how they're executed or reported to the providers? 

Do you think we can represent the presubmit behavior in these tests? How does the framework determine what attributes to include in the contract?

Reply:
Each set of consumer-provider interactions generate a pact in the form of a JSON file. So if HCA is a consumer and documents how it interacts with multiple vets-api endpoints, all of those will be grouped into one pact. And if there's another consumer, like GIBCT, its interactions will generate a separate pact. The main attributes to be concerned with are the same as any you would expect to describe a request or response. The format should be fairly clear from glancing at the [generated pact](pacts/va.gov-va.gov_api.json). The less obvious attributes are a result of using matchers, which generalize what the request or response body could look like using regex or other means. 

The provider will verify all at once (maybe there's a command to respond to specific consumers, but I don't know of it off the top of my head), but the verification results are reported separately for each individual consumer. The output look something like:

```
Verifying a pact between VA.gov and VA.gov API
  Given enrollment service is up
    an application for health care
      with POST /v0/health_care_applications
        returns a response which
          has status code 200
          has a matching body
          includes headers
            "Content-Type" which equals "application/json; charset=utf-8"
```

After getting more familiar with the flow of publishing pacts to a broker and verifying them, I think it might be more streamlined to describe consumers as the separate apps (HCA being one). VA.gov or vets-website as a whole could technically be a single consumer too, but the way I envision the process working is that changes would be made for a particular app, and that app would verify its contract with the API separately from changes to another app. That way, there can be concurrent pact verifications that don't get mixed up. This is all my understanding anyway.